qiskit_optimization/problems/quadratic_program.py

```diff
@@ -886,144 +858,38 @@ def maximize(
             self, constant, linear, quadratic, QuadraticObjective.Sense.MAXIMIZE
         )
 
-    def from_gurobipy(self, model: GurobiModel) -> None:
-        """Loads this quadratic program from a gurobipy model.
-
-        Note that this supports only basic functions of gurobipy as follows:
-        - quadratic objective function
-        - linear / quadratic constraints
-        - binary / integer / continuous variables
+    @staticmethod
+    def load(source: Any) -> "QuadraticProgram":
+        """Loads this quadratic program from an external source such as optimization models
+        and files supported by any of ``QuadraticProgramTranslator``.
 
         Args:
-            model: The gurobipy model to be loaded.
+            source: The external source to be loaded.
+
+        Returns:
+            the quadratic program of the translation from the source.
 
         Raises:
-            MissingOptionalLibraryError: gurobipy not installed
-            QiskitOptimizationError: if the model contains unsupported elements.
+            QiskitOptimizationError: if the provided source is not supported by any translator.
         """
-        if not _HAS_GUROBI:
-            raise MissingOptionalLibraryError(
-                libname="GUROBI",
-                name="GurobiOptimizer",
-                pip_install="pip install -i https://pypi.gurobi.com gurobipy",
-            )
-
-        # clear current problem
-        self.clear()
-
-        # Update the model to make sure everything works as expected
-        model.update()
-
-        # get name
-        self.name = model.ModelName
-
-        # get variables
-        # keep track of names separately, since gurobipy allows to have None names.
-        var_names = {}
-        for x in model.getVars():
-            if x.vtype == gp.GRB.CONTINUOUS:
-                x_new = self.continuous_var(x.lb, x.ub, x.VarName)
-            elif x.vtype == gp.GRB.BINARY:
-                x_new = self.binary_var(x.VarName)
-            elif x.vtype == gp.GRB.INTEGER:
-                x_new = self.integer_var(x.lb, x.ub, x.VarName)
-            else:
-                raise QiskitOptimizationError(
-                    "Unsupported variable type: {} {}".format(x.VarName, x.vtype)
-                )
-            var_names[x] = x_new.name
-
-        # objective sense
-        minimize = model.ModelSense == gp.GRB.MINIMIZE
+        return _load_qp_from_source(source)
 
-        # Retrieve the objective
-        objective = model.getObjective()
-        has_quadratic_objective = False
+    def export(self, translator: Optional[QuadraticProgramTranslator] = None) -> Any:
```

Comment:
Should we have `load/save`, `read/write`? A combination of `export/load` does not look naturally.

Reply:
This method basically returns a model of gurobi or docplex. Does `save` or `write` make sense?

Reply:
I removed `export` and reverted `to_docpelx` and `to_gurobipy`. I think it is easy for IDE to discover.

Reply:
Does it mean if we want to have another translator, we add a new method like `to_zzz` to QP?
Let's discuss namings separately as agreed.

Reply:
For example, pandas use such a design. https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_dict.html
Sure. I want to discuss the design and naming.